entrypoint.sh

```diff
@@ -7,227 +7,277 @@ set -o pipefail
+	local file="$1"
 
-  echo -e "\nRunning Super-Linter\n--------------------------\n"
-  echo "===============================" >> "$superlinter_logfile"
-  echo "SUPER-LINTER" >> "$superlinter_logfile"
+	if grep -q -v 'kustomize.config.k8s.io' "${file}" &&
+		grep -q -v "tekton" "${file}" &&
+		grep -q -E '(apiVersion):' "${file}" &&
+		grep -q -E '(kind):' "${file}"; then
+		return 1
+	fi
 
-  # Turn off the possum
-  export SUPPRESS_POSSUM="true"
-
-  if [[ -n ${GITHUB_WORKSPACE:-} ]]; then
-    echo "Setting ${GITHUB_WORKSPACE} as safe directory"
-    git config --global --add safe.directory "${GITHUB_WORKSPACE}"
+	return 0
+}
 
-  fi
+function detect_cloudformation_file() {
+	# Search for AWS Cloud Formation-related strings in files to determine which files to pass to cfn-lint
+	# Here a return of 0 indicates the function did not find a string match and exits with a success code,
+	# and 1 indicates the string was found. This more aligns with a boolean-like response.
+	local file="$1"
 
-  # When run in a pipeline, move per-repo configurations into the right location at runtime so super-linter finds them, overwriting the defaults.
-  # This will handle hidden and non-hidden files, as well as sym links
-  cp -p "${GITHUB_WORKSPACE:-.}/.github/linters/"* "${GITHUB_WORKSPACE:-.}/.github/linters/".* /etc/opt/goat/ || true
-  
-  set +e
-  superlinter_result=$(/action/lib/linter.sh >> "$superlinter_logfile" 2>&1; echo $?)
-  set -e
+	# Searches for a string specific to AWS CF templates
+	if grep -q 'AWSTemplateFormatVersion' "${file}" >/dev/null; then
+		return 1
+	fi
 
-  echo "-------------------------------" >> "$superlinter_logfile"
+	# Search for AWS, Alexa Skills Kit, or Custom Cloud Formation syntax within the file
+	if grep -q -E '(AWS|Alexa|Custom)::' "${file}" >/dev/null; then
+		return 1
+	fi
 
-  if [ "$superlinter_result" -gt 0 ]; then
-    cat "$superlinter_logfile"
-    linter_failed="true"
-    linter_failures+=("super-linter")
-  else
-    linter_successes+=("super-linter")
-  fi
+	return 0
 }
 
 function get_files_matching_filetype() {
-  local filetype="$1"
-  shift
-  local filenames=("$@")
-  matching_files=()
-
-  for file in "${filenames[@]}"; do
-    filename=$(basename "$file")
-    
-    if [[ "$filename" == *"$filetype" ]]; then
-      matching_files+=("$file")
-    fi
-  done
-  echo "${matching_files[@]}"
+	local filenames=("$@")
+	declare -a matching_files=()
+
+	for filetype in "${linter_filetypes[@]}"; do
+		for file in "${filenames[@]}"; do
+			filename=$(basename "${file}")
+			if [[ $filename == *"$filetype" ]]; then
+				if [ "${linter[name]}" == "cfn-lint" ]; then
+					if detect_cloudformation_file "${file}"; then
+						continue
+					fi
+				fi
+				if [ "${linter[name]}" == "kubeconform" ]; then
+					if detect_kubernetes_file "${file}"; then
+						continue
+					fi
+				fi
+				if [ "${linter[name]}" == "actionlint" ]; then
+					local action_path="./.github/workflows/"
```

Comment:
Shouldn't this be:
```suggestion
					local action_path="${GITHUB_WORKSPACE:-.}/.github/workflows/"
```

Reply:
It will default to ./ but also use `${GITHUB_WORKSPACE}` if it's set, such as in a pipeline, and has the added benefit that it's an absolute path when in a pipeline and doesn't care about other `cd`s or `pushd`s

Reply:
Done

Reply:
Done